File: models/pengaduan_timeline.go
package models

import "time"

type PengaduanTimeline struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	PengaduanId uint      `json:"pengaduan_id"`
	CreatedById uint      `json:"created_by_id"`
	Judul       *string   `json:"judul"`
	Keterangan  *string   `json:"keterangan"`
	CreatedAt   time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt   time.Time `gorm:"autoUpdateTime" json:"updated_at"`

	CreatedBy *User      `gorm:"foreignKey:CreatedById" json:"created_by"`
	Pengaduan *Pengaduan `gorm:"foreignKey:PengaduanId" json:"Pengaduan"`
}

File: app/src/main/java/com/example/sipamas_android/data/model/PengaduanTimeline.kt
package com.example.sipamas_android.data.model

data class PengaduanTimeline(
    val id: Int? = null,
    val pengaduan_id: Int? = null,
    val created_by_id: Int? = null,
    val judul: String? = null,
    val keterangan: String? = null,
    val created_at: String? = null,
    val updated_at: String? = null,
    val created_by: User? = null
)